packages/web/src/components/track/ViewerActionButtons.tsx

```diff
@@ -41,29 +41,102 @@ type ViewerActionButtonProps = {
   onClickGatedUnlockPill?: (e: MouseEvent) => void
 }
 
-const useEntityDetails = (contentId: ID, type: 'track' | 'collection') => {
-  const { data: track } = useGetTrackById({ id: contentId })
-  const { data: collection } = useGetPlaylistById({ playlistId: contentId })
+type EntityDetails = {
+  streamConditions: Nullable<AccessConditions>
+  isUnlisted: boolean
+  isFavorited: boolean
+  isReposted: boolean
+}
+
+const useTrackEntityDetails = (id: ID): EntityDetails => {
+  const { data: track } = useGetTrackById({ id })
 
   const {
     stream_conditions: streamConditions,
     has_current_user_saved: isFavorited,
     has_current_user_reposted: isReposted
-  } = (type === 'track' ? track : collection) ?? {}
+  } = track ?? {}
 
-  const isUnlisted =
-    type === 'track' ? track?.is_unlisted : collection?.is_private
+  const isUnlisted = track?.is_unlisted
   return {
-    streamConditions,
-    isUnlisted,
-    isFavorited,
-    isReposted
+    streamConditions: streamConditions ?? null,
+    isUnlisted: isUnlisted ?? false,
+    isFavorited: isFavorited ?? false,
+    isReposted: isReposted ?? false
+  }
+}
+
+const useCollectionEntityDetails = (id: ID): EntityDetails => {
+  const { data: collection } = useGetPlaylistById({ playlistId: id })
+
+  const {
+    stream_conditions: streamConditions,
+    has_current_user_saved: isFavorited,
+    has_current_user_reposted: isReposted
+  } = collection ?? {}
+
+  const isUnlisted = collection?.is_private
+  return {
+    streamConditions: streamConditions ?? null,
+    isUnlisted: isUnlisted ?? false,
+    isFavorited: isFavorited ?? false,
+    isReposted: isReposted ?? false
   }
 }
 
 export const ViewerActionButtons = ({
-  contentId,
   contentType,
+  ...rest
+}: ViewerActionButtonProps & { contentType: 'track' | 'collection' }) => {
+  return contentType === 'track' ? (
+    <TrackViewerActionButtons {...rest} />
+  ) : (
+    <CollectionViewerActionButtons {...rest} />
+  )
+}
+
+const TrackViewerActionButtons = ({
```

Comment:
nice! ultimately might wanna split these out into their own files. i like that they compose down to the same component in the end. this is a good pattern.

Reply:
yeah, i wanted to just keep the file hierarchy here the same for this PR at least, but not opposed to it. the problem was that every time we were fetching both a track and a playlist when we meant to be fetching only one